lib/ansible/module_utils/facts/system/pkg_mgr.py

```diff
@@ -82,7 +82,10 @@ def _check_rh_versions(self, pkg_mgr_name, collected_facts):
                 # just default to dnf
                 pkg_mgr_name = 'dnf'
         elif collected_facts['ansible_distribution'] == 'Amazon':
-            pkg_mgr_name = 'yum'
+            if int(collected_facts['ansible_distribution_version']) < 2022:
```

Comment:
this at least needs to be made more robust to a version that's not parseable as an `int`; either catching the exception that's raised and warning/falling back to another behavior or using something like `LooseVersion` for comparison... eg, what happens when they later set the version to `2022.1`?

Reply:
You are right @nitzmahone, I updated accordingly:

* switched over to `ansible_distribution_major_version` as to avoid `2022.1` - and also currently AL1 reports as `2018.03`
* added try-catch to cater for any weird other type of version. In those cases I would default to `dnf` as to future proof